src/connection.c

```diff
@@ -318,12 +325,26 @@ static void handleResolved(void *arg)
         goto fail;
     }
 
+
 #ifdef HAVE_TLS
     if (conn->rr->config->tls_enabled) {
-        if (redisInitiateSSLWithContext(&conn->rc->c, conn->rr->ssl) != REDIS_OK) {
-            CONN_LOG_ERROR(conn, "SSL Error!");
+        if (conn->ssl) {
+//            SSL_free(conn->ssl);
+            conn->ssl = NULL;
+        }
+
+        SSL *ssl = SSL_new(conn->rr->ssl);
+        if (!ssl) {
+            RedisModule_Log(conn->rr->ctx, REDIS_WARNING, "Couldn't create SSL object");
+            goto fail;
+        }
+        int result = redisInitiateSSL(&conn->rc->c, ssl);
+        if (result != REDIS_OK) {
+//            SSL_free(ssl);
```

Comment:
Taking a look at `redisInitiateSSL()`, it calls `SSL_free()` for some error cases but not all :( 
- OOM: It does not call `SSL_free()` on OOM. We can ignore this though.
- If the connection is already associated with a `redisSSL`, it doesn't call `SSL_free()`. 

It's not clear to me what we should do at this point. I believe this is a bug in hiredis API. Ownership is not clear. I also see other error paths inside hiredis this would be leaked. Maybe I'm missing something.

Reply:
https://github.com/RedisLabs/redisraft/blob/5e90f4437ade2a3cfb0f35f4fdcf3e2770bc162f/deps/hiredis/ssl.c#L315

Hmm, we added this line. Probably, we should also add `SSL_free()` to other error cases up above.

Reply:
yes, we should probably revert our change and fix hiredis / pull in updates hiredis.

Reply:
Damn, https://github.com/redis/hiredis/pull/939. 

They took this change back later, we didn't see it and merged that line into our hiredis version. So, you're right. I think we need to revert the change and call `SSL_free()` if this function fails.

Reply:
After rebasing to https://github.com/RedisLabs/redisraft/commit/dc1752ea467fe0328fd4dc0f88d9cf4b07ff1336
We can uncomment this line I believe.